### PR TITLE
game: Enable network menu in debug mode only

### PR DESCRIPTION
**Pull request by mrannanj**

### CODE REVIEW

File: src/game/scenes/mainmenu.c [hunk 1]
```diff
@@ -138,6 +138,7 @@ int mainmenu_create(scene *scene) {
     scene_set_dynamic_tick_cb(scene, mainmenu_tick);
     scene_set_startup_cb(scene, mainmenu_startup);
 
+#ifdef DEBUGMODE
     if(scene->gs->net_mode == NET_MODE_CLIENT) {
         component_action(guiframe_find(local->frame, NETWORK_BUTTON_ID), ACT_PUNCH);
         component_action(guiframe_find(local->frame, NETWORK_CONNECT_BUTTON_ID), ACT_PUNCH);
```
Comment (Nopey) [inline src/game/scenes/mainmenu.c:144]: Please do not break the net_mode field, this is set via commandline options that this PR is not removing.
I think leaving the network game launch options in is fine, as most users will not discover them-- and they will probably be useful when we need to check if netplay works with CI pipeline-built binaries.

File: src/game/scenes/mainmenu/menu_main.c [hunk 2]
```diff
@@ -105,9 +105,11 @@ component *menu_main_create(scene *s) {
     menu_attach(menu, textbutton_create(&tconf, "ONE PLAYER GAME", NULL, COM_ENABLED, mainmenu_1v1, s));
     menu_attach(menu, textbutton_create(&tconf, "TWO PLAYER GAME", NULL, COM_ENABLED, mainmenu_1v2, s));
     menu_attach(menu, textbutton_create(&tconf, "TOURNAMENT PLAY", NULL, COM_ENABLED, mainmenu_mechlab, s));
+#ifdef DEBUGMODE
     component *net = textbutton_create(&tconf, "NETWORK PLAY", NULL, COM_ENABLED, mainmenu_enter_network, s);
     widget_set_id(net, NETWORK_BUTTON_ID);
     menu_attach(menu, net);
+#endif
```
Comment (Nopey) [inline src/game/scenes/mainmenu/menu_main.c:112]: Maybe worth putting an `#else` with a disabled "NETWORK PLAY" button, similar to the "HELP" menu we haven't implemented yet?

**Review by Nopey — CHANGES_REQUESTED** [review]

File: src/game/scenes/mainmenu.c (hunk 1 above)
Comment (mrannanj) [reply src/game/scenes/mainmenu.c:144]: Okay, I let the `scene->gs->net_mode = NET_MODE_NONE;` stay, but it doesn't affect anything, as it is 0 anyway. About the CI build, that uses debug mode for testing, so don't need it there..

Comment (mrannanj) [reply src/game/scenes/mainmenu.c:144]: Oh yeah, and disabled the command line stuff too here.

File: src/game/scenes/mainmenu/menu_main.c (hunk 2 above)
Comment (mrannanj) [reply src/game/scenes/mainmenu/menu_main.c]: Added the disabled network play item.

**Review by Nopey — APPROVED** [review]
Unsure what the motivation for removing the launch options is (`--connect` and `--port`), but the actual code changes to accomplish this are fine.

I think leaving the network game launch options in is a good idea, as they will probably be useful when we need to check if netplay works with CI pipeline-built binaries. (and users are unlikely to discover them in error)

**Review by Nopey — APPROVED** [review]

**Review by Vagabond — APPROVED** [review]
I haven't tested this but it looks correct.

**Pull request merged**